src/wasm/wasm-stack.cpp

```diff
@@ -1686,7 +1692,8 @@ void BinaryInstWriter::visitHost(Host* curr) {
 }
 
 void BinaryInstWriter::visitRefNull(RefNull* curr) {
-  o << int8_t(BinaryConsts::RefNull);
+  o << int8_t(BinaryConsts::RefNull)
+    << binaryHeapType(curr->type.getHeapType());
```

Comment:
Binary variant of `(ref.null extern)` etc. This is probably wrong because the proposal states that these are `s33`s?

Reply:
Yes, technically s33, though that's just so the full 32-bit index range can be represented. It's probably OK for now to use s32 and have a TODO.